cmd/subnetcmd/join.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+	"github.com/spf13/cobra"
+)
+
+var (
+	avagoConfigPath string
+	print           bool
+)
+
+// avalanche subnet deploy
+func newJoinCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "join [subnetName]",
+		Short: "Instruct a validator node to begin validating a new subnet",
+		Long: `Instruct a validator node to begin validating a new subnet.
+The NodeID of that validator node must have been whitelisted by one of the 
+subnet's control keys for this to work.
+
+The node also needs to be restarted.
+If --avalanchego-config is provided, this command tries to edit the config file at that path
+(consider correct permissions).`,
+		RunE: joinCmd,
+		Args: cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&avagoConfigPath, "avalanchego-config", "", "file path of the avalanchego config file")
+	cmd.Flags().BoolVar(&print, "print", false, "if true, print the manual config without prompting")
+	return cmd
+}
+
+func joinCmd(cmd *cobra.Command, args []string) error {
+	if print && avagoConfigPath != "" {
+		return errors.New("--print and --avalanchego-config simultaneously is not supported")
+	}
+
+	/*
+		chains, err := validateSubnetName(args)
+		if err != nil {
+			return err
+		}
+		subnetName := chains[0]
+
+		sc, err := app.LoadSidecar(subnetName)
+		if err != nil {
+			return err
+		}
+
+		subnetID := sc.SubnetID
+		if subnetID == ids.Empty {
+			return errNoSubnetID
+		}
+	*/
+	subnetIDstr := "29Zd5yhP7Yb2cTebBbUVKUjHHNviHzgj1y9kKJsvMn2dyWnkpG"
+	var network models.Network
+	networkStr, err := prompts.CaptureList(
+		"Choose a network to deploy on (this command only supports public networks)",
+		[]string{models.Fuji.String(), models.Mainnet.String()},
+	)
+	if err != nil {
+		return err
+	}
+	network = models.NetworkFromString(networkStr)
+	networkLower := strings.ToLower(network.String())
+
+	ask := "Would you like to check if your node is already whitelisted to join this subnet?"
+	yes, err := prompts.CaptureYesNo(ask)
+	if err != nil {
+		return err
+	}
+	if yes {
+		isWhitelisted, err := checkIsWhitelisted(ids.GenerateTestID(), network)
```

Comment:
I think you can already pass a subnet id to it, instead of GenerasteTestID

Reply:
Yes, but because the actual subnetID is commented out above, I am just using a test ID until the code can be enabled (it's in the addValidator PR which itself can't be merged yet...)